### DISCUSSION TITLE
Problems setting up development environment

### DISCUSSION THREAD

**Discussion by jokakilla** [blakeblackshear/frigate#3987] [Ask A Question]
I'm creating this as I don't wanted to add more questions https://github.com/blakeblackshear/frigate/issues/3981 here which are not related to the issue.

I've followed the guide for setting up the dev env -> https://docs.frigate.video/contributing
Installed all prerequisites:
git clone https://github.com/blakeblackshear/frigate.git
joka@joka:~$ code --version
1.71.2
joka@joka:~$ make --version
GNU Make 4.2.1
joka@joka:~$ docker --version
Docker version 20.10.18, build b40c2f6
joka@joka:~$ docker compose version
Docker Compose version v2.10.2
joka@joka:~$ node --version
v14.20.1

Source code is located at /home/joka/frigate so basically inside my home directory.

Calling make failed because web/dist could not be copied to web.
Advisory in the comment https://github.com/blakeblackshear/frigate/issues/3981 was to run make web first.

I guess this is make build_web (according to the make file)?
make build_web results in error 243.
```
docker run --volume /home/joka/frigate/web:/web -w /web --volume /etc/passwd:/etc/passwd:ro --volume /etc/group:/etc/group:ro -u 1000:1000 node:16 /bin/bash -c "npm install && npm run build"

make: *** [Makefile:12: build_web] Fehler 243
```

 A search for this issue lead to stackoverflow. Someone mentioned this is an issue with the node:16 image. Solution: Take node:16.14.2
With 16.14.2 it did more than before but result was tons of permission denied errors:
```
....
npm WARN old lockfile FetchError: Invalid response body while trying to fetch https://registry.npmjs.org/unbox-primitive: EACCES: permission denied, mkdir '/home/joka'
npm WARN old lockfile     at /usr/local/lib/node_modules/npm/node_modules/minipass-fetch/lib/body.js:162:15
npm WARN old lockfile     at async Array.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:759:9)
npm WARN old lockfile  Could not fetch metadata for unbox-primitive@1.0.1 FetchError: Invalid response body while trying to fetch https://registry.npmjs.org/unbox-primitive: EACCES: permission denied, mkdir '/home/joka'
npm WARN old lockfile     at /usr/local/lib/node_modules/npm/node_modules/minipass-fetch/lib/body.js:162:15
npm WARN old lockfile     at async Array.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:759:9) {
npm WARN old lockfile   code: 'EACCES',
npm WARN old lockfile   errno: 'EACCES',
npm WARN old lockfile   syscall: 'mkdir',
npm WARN old lockfile   path: '/home/joka',
npm WARN old lockfile   type: 'system'
npm WARN old lockfile }
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /home/joka
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/home/joka'
npm ERR!  [Error: EACCES: permission denied, mkdir '/home/joka'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/home/joka'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

```

Any help appreciated!

**Comment by NickM-27** [comment]
Where are you running make from? (Inside a dev container or on the host?)

I usually just build web from the host if the make doesn't work by running `cd web && npm install && npm run build` and then make works as expected

**Comment by jokakilla** [comment]
Very nice...that did the trick. Everything else (make, start vs code, run python3 -m frigate) was working fine then. 
Frigate is reachable on http://localhost:5000 and streams my prerecorded mp4.

Thanks for your help so far :)

**Comment by tpjanssen** [comment]
I'm having the same issue.

Another thing, in the make file "nginx_frigate" is specified, but the file "docker/Dockerfile.nginx" does not exist in this branch?

**Reply by NickM-27** [reply]
You don't need to make that as it's pulled from remote. make nginx_frigate is not called anywhere

**Reply by tpjanssen** [reply]
Thanks for the quick reply. And what about fehler 243? This error is thrown when running from a dev container...

**Reply by NickM-27** [reply]
Please include a screenshot or full paste of the error

**Reply by tpjanssen** [reply]
> Please include a screenshot or full paste of the error

A screenshot of the full error:

![image](https://user-images.githubusercontent.com/25168870/193748640-ee472ce2-4046-4f4b-96cd-5c8771112bbb.png)

Thanks for helping out.

**Reply by jokakilla** [reply]
Did you try it without docker?
cd web && npm install && npm run build
was working fine for me. This Error 243 was somehow coming from the node docker container.

**Reply by tpjanssen** [reply]
I'm running within docker, on a Windows machine.

**Reply by NickM-27** [reply]
Have you tried running `make build_web` ?
